src/ja/common/proxy/command_handler.py

```diff
@@ -45,20 +45,28 @@ def _listen(self) -> None:
         while True:
             connection, client_address = named_socket.accept()
             try:
-                input_string = ""
-                while True:
-                    data = connection.recv(1024)
-                    if data[-1:] == b"\0":
-                        input_string += data[:-1].decode()
-                        break
-                    input_string += data.decode()
+                command_bytes = bytes(0)  # First 8 bytes encode command length
+                command_length = None
+                while command_length is None or len(command_bytes) < command_length:
+                    command_bytes += connection.recv(1024)
```

Comment:
I think it would have been much simpler if you just read the first 8 bytes with connection.recv(8-len(received)) and then another loop for the other bytes but that's inconsequential.